relay-event-normalization/src/normalize/nel.rs

```diff
@@ -40,22 +223,202 @@ pub fn create_log(nel: Annotated<NetworkReportRaw>, received_at: DateTime<Utc>)
         }};
     }
 
-    add_attribute!("todo.nel.referrer", body.referrer);
-    add_attribute!(
-        "todo.nel.server_ip",
-        body.server_ip.map_value(|s| s.to_string())
-    );
-    add_attribute!("todo.nel.elapsed_time", body.elapsed_time);
-    add_attribute!("todo.nel.error_type", body.ty);
-    add_attribute!("todo.nel.phase", body.phase.map_value(|s| s.to_string()));
-    add_attribute!("todo.nel.sampling_fraction", body.sampling_fraction);
+    macro_rules! add_string_attribute {
+        ($name:literal, $value:expr) => {{
+            let val = $value.to_string();
+            if !val.is_empty() {
+                attributes.insert($name.to_owned(), val);
+            }
+        }};
+    }
+
+    add_string_attribute!("sentry.origin", "auto.http.browser_report.nel");
+    add_string_attribute!("browser.report.type", "network-error");
+
+    // Handle URL and extract server address if available
+    add_attribute!("url.full", raw_report.url.clone());
+    if let Some(url_str) = raw_report.url.value() {
+        let url_domain = extract_server_address(url_str);
+        add_string_attribute!("url.domain", &url_domain);
+    }
```

Comment:
Untested but if you swap the order, we should be able to get away without a clone:

```suggestion
    // Handle URL and extract server address if available
    if let Some(url_str) = raw_report.url.value() {
        let url_domain = extract_server_address(url_str);
        add_string_attribute!("url.domain", &url_domain);
    }
    add_attribute!("url.full", raw_report.url);
```

Reply:
Ah! That makes sense!